### DISCUSSION TITLE
How to migrate a bash based barebone restic setup to "resticprofile"; keeping the same sources and remote b2 bucket?

### DISCUSSION THREAD

**Discussion by amarendra** [creativeprojects/resticprofile#273] [Q&A]
This is how I [backed up](https://github.com/amarendra/.dotfiles/blob/master/scripts/restic.backup.sh) and [maintained](https://github.com/amarendra/.dotfiles/blob/master/scripts/restic.maintain.sh) using `restic` on my last Mac (Catalina) in this very crude way.

And use [these](https://github.com/amarendra/.dotfiles/blob/master/LaunchAgents/amar.restic.backup.plist) for [scheduling](https://github.com/amarendra/.dotfiles/blob/master/LaunchAgents/amar.restic.maintenance.plist). 

I was hoping whether I could simplify all this using `resticprofile` on my new Mac (Sonoma) keeping same:
- existing remote - backblaze b2 bucket
- same remote url and repo password etc (I have them in a local `restic.env` file that I'd load when using and preferably unload after using)
- restic encryption passphrase
- same source set (of course) - I have them listed in a separate `RESTIC_INCLUDE_FILE` file. I can move it directly to YAML if needed.

Basically start appending to the same remote b2 bucket; only diff; and not a fresh backup. I already have all the local files I had in source set on my Mac.

I can start at https://creativeprojects.github.io/resticprofile/configuration/examples/ but before that I wanted to see whether the communicaty has some more closer examples or tutorial/blogs for what I want to achieve.

**Comment by amarendra** [comment]
Or something like https://github.com/cupcakearmy/autorestic is better suited for my use case? (Or maybe even https://github.com/rustic-rs/rustic?)

I noticed restic forum admin to recommend at resticprofile couple of times and it has a mention [here](https://restic.readthedocs.io/en/latest/040_backup.html#scheduling-backups); so that was definitely an encouragement :)

**Reply by creativeprojects** [reply]
Can you post your scripts in a message? The links you sent are probably from a private repository: https://github.com/amarendra/.dotfiles/ and looking by the name it's for the best 😆 

Apart from that `autorestic` and `resticprofile` are very similar. I didn't know the other existed before I started `resticprofile`. The biggest difference is that `resticprofile` schedules jobs for you from the configuration, you don't need to do it manually (which is not a big deal anyway).

**Reply by amarendra** [reply]
Damn :) And you are right. Even though these files have absolutely zero potential security issue phrases I had kept them private that even if by mistake I push something they at least do not become public.

I am uploading the files here (with just a folder name change which could have been left unedited really).

restic:
[restic.backup.sh.txt](https://github.com/creativeprojects/resticprofile/files/13184671/restic.backup.sh.txt)
[restic.maintain.sh.txt](https://github.com/creativeprojects/resticprofile/files/13184674/restic.maintain.sh.txt)

launch agents:
[amar.restic.backup.plist.txt](https://github.com/creativeprojects/resticprofile/files/13184678/amar.restic.backup.plist.txt)
[amar.restic.maintenance.plist.txt](https://github.com/creativeprojects/resticprofile/files/13184679/amar.restic.maintenance.plist.txt)

> Apart from that autorestic and resticprofile are very similar.

Oh. resticprofile seems more actively developed but that might just be coincidence. They both are written in Go. I would assume someday these might join forces or might be base for a nice little simple GUI as well :)

> resticprofile schedules jobs for you from the configuration, you don't need to do it manually

But to make things automated I guess that is kinda a decently size difference, right?

**Reply by jkellerer** [reply]
The following `profiles.yaml` should come close to your current setup:

```yaml
# yaml-language-server: $schema=https://creativeprojects.github.io/resticprofile/jsonschema/config-2.json

version: "2"

profiles:
  default:
    repository: "..."
    password-file: "..."
    env:
      B2_ACCOUNT_ID: "..."
      B2_ACCOUNT_KEY: "..."
  
  # backup profile
  system:
    inherit: default
    backup:
      check-after: true
      
      schedule: 
        - "08:00"
        - "12:00"
        - "16:00"
        - "22:00"
      schedule-lock-wait: 15m
      schedule-log: "/Users/amar/Library/Logs/restic/restic-backup.log"

      # may be optional with files-from
      source: 
        - /path/to/backup
        - /another/path/to/backup

      files-from: /path/to/file/with/source-patterns

      exclude-caches: true
      exclude-file: /path/to/file/with/exclude-patterns

    # retention (runs forget after backup)
    retention:
      path: false
      keep-last: 10
      keep-daily: 7
      keep-monthly: 12
      keep-yearly: 3

    # physically remove data and compact repo
    prune:
      schedule: Wed,Fri 13:30
      schedule-lock-wait: 15m
      schedule-log: "/Users/amar/Library/Logs/restic/restic-prune.log"

  # full check profile
  fullcheck:
    inherit: default
    check:
      schedule: monthly
      schedule-lock-wait: 15m
      schedule-log: "/Users/amar/Library/Logs/restic/restic-full-check.log"

      read-data: true
```

Usage (assuming `restic`, `resticprofile` and `profiles.yaml` is in the correct paths):

```
➜ ./resticprofile --dry-run backup
2023/10/27 17:06:48 using configuration file: profiles.yaml
2023/10/27 17:06:48 profile 'system': starting 'backup'
2023/10/27 17:06:48 dry-run: restic backup --exclude-caches --exclude-file=/path/to/file/with/exclude-patterns --files-from=/path/to/file/with/source-patterns --host=MYHOST01 --password-file=××× --repo=××× /path/to/backup /another/path/to/backup
2023/10/27 17:06:48 profile 'system': finished 'backup'
2023/10/27 17:06:48 profile 'system': cleaning up repository using retention information
2023/10/27 17:06:48 dry-run: restic forget --host=MYHOST01 --keep-daily=7 --keep-last=10 --keep-monthly=12 --keep-yearly=3 --password-file=××× --repo=×××
2023/10/27 17:06:48 profile 'system': checking repository consistency
2023/10/27 17:06:48 dry-run: restic check --password-file=××× --repo=×××

➜ ./resticprofile --dry-run prune 
2023/10/27 17:06:53 using configuration file: profiles.yaml
2023/10/27 17:06:53 profile 'system': starting 'prune'
2023/10/27 17:06:53 dry-run: restic prune --password-file=××× --repo=×××
2023/10/27 17:06:53 profile 'system': finished 'prune'

➜ ./resticprofile --dry-run fullcheck.check
2023/10/27 17:06:57 using configuration file: profiles.yaml
2023/10/27 17:06:57 profile 'fullcheck': starting 'check'
2023/10/27 17:06:57 dry-run: restic check --password-file=××× --read-data --repo=×××
2023/10/27 17:06:57 profile 'fullcheck': finished 'check'
```

To schedule / unschedule run:

```
resticprofile system.schedule
resticprofile system.unschedule

resticprofile fullcheck.schedule
resticprofile fullcheck.unschedule
```
